src/ecdsa/mod.rs

```diff
@@ -68,24 +72,120 @@ impl Signature {
     }
 }
 
+/// The arguments used to derive randomness used for presignature rerandomization.
+/// Presignature rerandomization has been thoroughly described in
+/// [GS21] https://eprint.iacr.org/2021/1330.pdf
+///
+/// *** Warning ***
+/// Following [GS21] https://eprint.iacr.org/2021/1330.pdf, the entropy should
+/// be public, freshly generated, and unpredictable.
+pub struct RerandomizationArguments<'a> {
+    pub pk: &'a AffinePoint,
+    pub msg_hash: &'a Scalar,
+    pub big_r: &'a AffinePoint,
+    pub participants: &'a ParticipantList,
+    /// Fresh, Unpredictable, and Public source of entropy
+    pub entropy: [u8; 32],
+}
+
+impl<'a> RerandomizationArguments<'a> {
+    /// The following salt is picked by hashing with sha256
+    /// "NEAR 6.4478$ 7:20pm CEST 2024-11-24"
+    /// Based on [Krawczyk10] paper:
+    /// ``[...] in most applications the extractor key (or salt) can be used
+    /// repeatedly with many (independent) samples from the same source [...]''
+    const SALT: [u8; 32] = [
+        0x32, 0x8a, 0x47, 0xc2, 0xb8, 0x79, 0x44, 0x45, 0x25, 0x5c, 0x16, 0x47, 0x60, 0x8d, 0xf5,
+        0xdb, 0x85, 0xc6, 0x8b, 0xb0, 0xe7, 0x17, 0x0a, 0xbe, 0xc5, 0x34, 0xdf, 0x27, 0x64, 0xa4,
+        0x58, 0x31,
+    ];
+
+    pub fn new(
+        pk: &'a AffinePoint,
+        msg_hash: &'a Scalar,
+        big_r: &'a AffinePoint,
+        participants: &'a ParticipantList,
+        entropy: [u8; 32],
+    ) -> RerandomizationArguments<'a> {
+        RerandomizationArguments {
+            pk,
+            msg_hash,
+            big_r,
+            participants,
+            entropy,
+        }
+    }
+
+    /// Derives a random string from the public key, message hash, presignature R,
+    /// set of participants and the entropy.
+    ///
+    /// Outputs a random string computed as HKDF(entropy, pk, hash, R, participants)
+    pub fn derive_randomness(&self) -> Scalar {
+        // create a string containing (pk, msg_hash, big_r, sorted(participants))
+        let pk_encoded_point = self.pk.to_encoded_point(true);
+        let encoded_pk: &[u8] = pk_encoded_point.as_bytes();
+        let encoded_msg_hash: &[u8] = &self.msg_hash.to_bytes()[..];
```

Comment:
No need to slice here
```suggestion
        let encoded_msg_hash: &[u8] = &self.msg_hash.to_bytes();
```

Reply:
Outdated after [f3d7b6b](https://github.com/near/threshold-signatures/pull/86/commits/f3d7b6b41867deaa635e0ca6681bcde5901b5060)